testing/cli/test_sct_run_batch.py

```diff
@@ -12,46 +13,43 @@
 from spinalcordtoolbox.scripts import sct_run_batch
 
 
-def write_dummy_script(script):
-    """
-    Dummy script that displays subject
-    :param script: NamedTemporaryFile object. Script to write
-    :return:
-    """
+@pytest.fixture
+def dummy_script(tmp_path):
+    """Dummy executable script that displays subject."""
+    path_out = str(tmp_path / "dummy_script.sh")
     script_text = """
     #!/bin/bash
     SUBJECT=$1
     echo $SUBJECT
     """
-    # indexing removes beginning newline
-    script.write(dedent(script_text)[1:])
-    script.flush()
+    with open(path_out, 'w') as script:
+        # indexing removes beginning newline
+        script.write(dedent(script_text)[1:])
+        script.flush()
```

Comment:
Is flushing still necessary, given the `with open()` context?

Reply:
[Definitely not.](https://stackoverflow.com/a/51300280) Good catch. :)